playbooks/prerun-upgrade-checks.yml

```diff
@@ -42,6 +42,31 @@
               acs_play_repository_api_explorer_version != installed_components.api_explorer.version
 
         - name: Fail on unsupported AMPS upgrades
+          vars:
+            amps: >-
```

Comment:
you can make it simpler by using an accumulator pattern, e.g. define an empty fact `[]`, and then have more tasks which overwrite the previous fact and using `combine` to add one more entry (eventually with conditionals)